src/shim.rs

```diff
@@ -57,23 +57,35 @@ macro_rules! impl_conditional_statement {
             use crate::{ConditionalStatement, SimpleExpr, IntoCondition, $struct_name};
 
             impl $struct_name {
-                pub fn and_where(&mut self, other: SimpleExpr) -> &mut Self {
-                    <Self as ConditionalStatement>::and_where(self, other)
```

Comment:
I think it's better to keep these as is. Then, calling `into` method when constructing filter with `LikeStatement`. E.g.

```rs
Query::select()
    .column(Char::Character)
    .from(Char::Table)
    .and_where(Expr::col(Char::Character).like("A").escape('\\').into())
```

Reply:
Api doesn't look good with: `.into()`.